hack/minikube-rook-mirror-setup.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+set -x
+set -e -o pipefail
+
+## Variables
+PRIMARY_CLUSTER="${PRIMARY_CLUSTER:-hub}"
+SECONDARY_CLUSTER="${SECONDARY_CLUSTER:-cluster1}"
+POOL_NAME="replicapool"
+
+## Usage
+usage()
+{
+	set +x
+	echo "Usage:"
+	echo "  $0"
+	echo "  Available environment variables:"
+	echo "    minikube primary cluster PRIMARY_CLUSTER ${PRIMARY_CLUSTER}"
+	echo "    minikube secondary cluster SECONDARY_CLUSTER ${SECONDARY_CLUSTER}"
+	exit 0
+}
+
+SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME=$(kubectl get cephblockpools.ceph.rook.io "${POOL_NAME}" --context="${SECONDARY_CLUSTER}" -nrook-ceph -o jsonpath='{.status.info.rbdMirrorBootstrapPeerSecretName}')
+SECONDARY_CLUSTER_SECRET=$(kubectl get secret -n rook-ceph "${SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME}" --context="${SECONDARY_CLUSTER}" -o jsonpath='{.data.token}'| base64 -d)
+SECONDARY_CLUSTER_SITE_NAME=$(kubectl get cephblockpools.ceph.rook.io "${POOL_NAME}" --context="${SECONDARY_CLUSTER}" -nrook-ceph -o jsonpath='{.status.mirroringInfo.site_name}')
+
+echo SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME is "$SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME"
+echo Token for the secondary cluster is "$SECONDARY_CLUSTER_SECRET"
+echo SECONDARY_CLUSTER_SITE_NAME is "${SECONDARY_CLUSTER_SITE_NAME}"
+
+kubectl -n rook-ceph create secret generic --context="${PRIMARY_CLUSTER}" "${SECONDARY_CLUSTER_SITE_NAME}" \
+        --from-literal=token="${SECONDARY_CLUSTER_SECRET}" \
+        --from-literal=pool="${POOL_NAME}"
+
+cat <<EOF | kubectl --context="${PRIMARY_CLUSTER}" apply -f -
+apiVersion: ceph.rook.io/v1
+kind: CephRBDMirror
+metadata:
+  name: my-rbd-mirror
+  namespace: rook-ceph
+spec:
+  count: 1
+  peers:
+    secretNames:
+      - "${SECONDARY_CLUSTER_SITE_NAME}"
+EOF
+
+kubectl wait deployments -n rook-ceph --for condition=available rook-ceph-rbd-mirror-a --timeout=60s --context="${PRIMARY_CLUSTER}"
+
+if kubectl --context="${PRIMARY_CLUSTER}" -n rook-ceph  get pods | grep -q rbd-mirror
+then
+        echo "Found rbd mirror pod"
+else
+        echo "Did not find rbd mirror pod, :("
+        exit 1
+fi
+
+if [[ $(kubectl get cephblockpools.ceph.rook.io replicapool --context="${PRIMARY_CLUSTER}" -nrook-ceph -o jsonpath='{.status.mirroringStatus.summary.daemon_health}') != "OK" ]]
```

Comment:
I have faced an issue here because it takes some time to reflect the status after deploying the mirroring pod. Can we just have a few seconds (5 or 10) delay here or check with for loop?

Reply:
Hmmm... I had a sleep at line 47 that I converted to a wait. We possibly need to loop here. Will do.

Reply:
Done